src/main/kotlin/com/cjbooms/fabrikt/generators/controller/ControllerGeneratorUtils.kt

```diff
@@ -48,4 +49,25 @@ object ControllerGeneratorUtils {
 
     private fun httpVerbMethodName(verb: String, isSingleResource: Boolean) =
         if (isSingleResource) "${verb}ById" else verb
+
+    enum class SecurityOption {
+        NO_SECURITY,
+        AUTHENTICATION_REQUIRED,
+        AUTHENTICATION_PROHIBITED,
+        AUTHENTICATION_OPTIONAL
+    }
```

Comment:
**suggestion:** It seems to me that rather often, we have the need to check "is this either REQUIRED or OPTIONAL" and "is this either PROHIBITED or OPTIONAL". So I would suggest to add respective fields to these enums.

```kotlin
    enum class SecurityOption(val allowsAuthorized: Boolean, val allowsAnonymous: Boolean) {
        NO_SECURITY(false, false),
        AUTHENTICATION_REQUIRED(true, false),
        AUTHENTICATION_PROHIBITED(false, true),
        AUTHENTICATION_OPTIONAL(true, true),
    }
```

Reply:
We need to check it once per generator, which I would not see as very often. But I can make the changes accordingly, as it makes the proposed code in the generator slimmer.